install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml

```diff
@@ -370,6 +375,28 @@ spec:
 {{- end }}
       hostNetwork: true
       initContainers:
+{{- if .Values.cgroup.autoMount.enabled }}
+      # Required to mount cgroup2 filesystem on the underlying Kubernetes node.
+      # We use nsenter command with host's cgroup and mount namespaces enabled.
+      - name: mount-cgroup
+        env:
+          - name: CGROUP_ROOT
+            value: {{ .Values.cgroup.hostRoot }}
+        command:
+          - nsenter
+          - --cgroup=/hostpid1ns/cgroup
+          - --mount=/hostpid1ns/mnt
+          - --
+          - sh
+          - -c
+          - 'mount | grep "$CGROUP_ROOT type cgroup2" || { echo "Mounting cgroup filesystem..."; mount -t cgroup2 none $CGROUP_ROOT; }'
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}{{ if .Values.image.useDigest }}@{{ .Values.image.digest }}{{ end }}"
```

Comment:
@aditighag btw just while looking back over this implementation I noticed that this container is missing `imagePullPolicy`. We shuold probably set it the same way as the other init container below.